src/client/kv_client.rs

```diff
@@ -88,50 +142,40 @@ impl KvClient {
                 }
             }
         }
-        Err(CollectionNotFound)
+        Err(CollectionNotFound { key: Some(key) })
     }
 
     async fn get_bucket_config(
-        &self,
+        seeds: Vec<String>,
         bucket: String,
+        tls_config: ClusterTlsConfig,
+        http_agent: &HTTPHandler,
         deadline: Instant,
         ctrl_c: Arc<AtomicBool>,
     ) -> Result<BucketConfig, ClientError> {
         let path = format!("/pools/default/b/{}", bucket);
         let mut final_error_content = None;
         let mut final_error_status = 0;
-        for seed in &self.seeds {
+        for seed in seeds {
             let host_split: Vec<String> = seed.split(':').map(|v| v.to_owned()).collect();
 
             let host: String;
             let port: i32;
             if host_split.len() == 1 {
                 host = seed.clone();
-                port = if self.tls_config.enabled() {
-                    18091
-                } else {
-                    8091
-                };
+                port = if tls_config.enabled() { 18091 } else { 8091 };
```

Comment:
so we have custom port support yet?

Reply:
I *think* that this if is the custom port support. Check out the else below. This block covers the case where the user didn't define a port so we use the defaults here.